pkg/cloud/config/controller.go

```diff
@@ -47,8 +48,13 @@ type Controller struct {
 
 // NewController initializes an Config Controller
 func NewController(cp models.Provider) *Controller {
-	providerConfig := provider.ExtractConfigFromProviders(cp)
-	watchers := GetCloudBillingWatchers(providerConfig)
+	var watchers map[ConfigSource]cloud.KeyedConfigWatcher
+	if env.IsKubernetesEnabled() {
```

Comment:
Let's get more reviews and see what others think?

Reply:
The ideal thing to do here would be to have a diagnostics page and endpoint, of which this would be one item, as well as the prometheus connection, cloud cost connections, etc, and that would be better suited to a separate issue.